File: fns-hl7-pipeline/fn-structural-validator/src/main/java/com/function/hl7/structure/validator/Function.scala
package com.function.hl7.structure.validator;

// import com.microsoft.azure.functions.ExecutionContext;
// import com.microsoft.azure.functions.HttpMethod;
// import com.microsoft.azure.functions.HttpRequestMessage;
// import com.microsoft.azure.functions.HttpResponseMessage;
// import com.microsoft.azure.functions.HttpStatus;
// import com.microsoft.azure.functions.annotation.AuthorizationLevel;
// import com.microsoft.azure.functions.annotation.FunctionName;
// import com.microsoft.azure.functions.annotation.HttpTrigger;

import com.microsoft.azure.functions.ExecutionContext
import com.microsoft.azure.functions.annotation.EventHubTrigger
import com.microsoft.azure.functions.annotation.FunctionName

import java.util.Optional;

import scala.util.{Try, Failure, Success}
import scala.concurrent.duration._
import java.util.concurrent.TimeUnit

import scala.concurrent._

import cdc.xlr.structurevalidator._;

/**
 * Azure Functions with HTTP Trigger.
 */
class Function {
    /**
     * This function listens at endpoint "/api/HttpExample"
     */
    @FunctionName("structuralvalidator001")
    def run(
        @EventHubTrigger(
          name = "msg",
          eventHubName = "eventhub002",
          connection = "EventHubConnectionString") 
          message: String,
          context: ExecutionContext): Unit = {

          context.getLogger.info("Scala event hub triggered a request.")
          context.getLogger.info("message: -> " + message)
        
    } // .EventHubTrigger
}

File: fns-hl7-pipeline/fn-message-processor/function/src/main/kotlin/rdmp/Function.kt
package com.example

import com.microsoft.azure.functions.ExecutionContext
import com.microsoft.azure.functions.annotation.EventHubTrigger
import com.microsoft.azure.functions.annotation.FunctionName

import com.google.gson.Gson 
import com.azure.storage.blob.*
import com.azure.storage.blob.models.*

import com.azure.messaging.eventhubs.*

import java.util.UUID
import java.io.*

const val BLOB_CREATED = "Microsoft.Storage.BlobCreated"
const val UTF_BOM = "\uFEFF"

/**
 * Azure Functions with Event Hub Trigger.
 */
class Function {
    @FunctionName("messageprocessor001")
    fun eventHubProcessor(
            @EventHubTrigger(
                name = "msg", 
                // TODO:
                eventHubName = "eventhub004",
                connection = "EventHubConnectionString") 
                message: String?,
            context: ExecutionContext) {

        context.logger.info("message: --> " + message)


    } // .eventHubProcessor

} // .Function

